src/couch_mrview/src/couch_mrview_util.erl

```diff
@@ -523,6 +525,33 @@ validate_args(#mrst{} = State, Args0) ->
     end.
 
 
+apply_limit(ViewPartitioned, Args) ->
+    LimitType = case ViewPartitioned of
+        true -> "partition_query_limit";
+        false -> "query_limit"
+    end,
+    
+    MaxLimit = config:get_integer("query_server_config",
+        LimitType, ?MAX_VIEW_LIMIT),
+
+    % set the highest limit possible if a user has not 
+    % specified a limit
+    Args1 = case Args#mrargs.limit =:= ?MAX_VIEW_LIMIT
+        andalso MaxLimit =/= infinity of
+        false ->
+            Args;
+        true ->
+            Args#mrargs{
+                limit = MaxLimit
+            }
+    end,
+
+    if Args1#mrargs.limit =< MaxLimit -> Args1; true ->
+        Fmt = "Limit is too large, must not exceed ~p",
```

Comment:
Something is wrong here but I'm not immediately seeing why. The view_multi_key_design.js test has failed with this error message on each worker. Perhaps we left a bad config from an unrelated test?

Reply:
I found the issue. It was because of the `fix_skip_and_limit` function https://github.com/apache/couchdb/blob/master/src/fabric/src/fabric_view.erl#L380

We made the max limit greater than the defined max limit.

Reply:
I see the issue there but I'm not seeing what changed to account for it.